source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -715,6 +715,20 @@ namespace
                 else
                 {
                     destPort = destNodeDef->addInput(attr->getName(), attr->getType().str());
+
+                    const vector<RtToken>& metadataNames = input->getMetadataOrder();
+                    for (auto metadataName : metadataNames)
+                    {
+                        const RtTypedValue* metadataValue = input->getMetadata(metadataName);
+                        if (metadataValue)
+                        {
+                            std::string metadataString = metadataValue->getValueString();
+                            if (!metadataString.empty())
+                            {
+                                destPort->setAttribute(metadataName, metadataString);
```

Comment:
About this loop to write metadata, I think the call to `writeMetadata(attr, destPort, attrMetadata);` below (line 754) should already handle writing this. Is it not?